alibi_detect/cd/classifier.py

```diff
@@ -26,12 +29,14 @@ def __init__(
             seed: int = 0,
             optimizer: Optional[Callable] = None,
             learning_rate: float = 1e-3,
-            compile_kwargs: Optional[dict] = None,
             batch_size: int = 32,
+            preprocess_batch_fn: Optional[Callable] = None,
             epochs: int = 3,
             verbose: int = 0,
             train_kwargs: Optional[dict] = None,
             device: Optional[str] = None,
+            dataset: Optional[Callable] = None,
```

Comment:
Is it definitely necessary to pass this `dataset` argument? If so I think it would be helpful to make description clearer as to when it is useful and that's it's just a wrapper rather something containing actual data.

Reply:
It is necessary and should hopefully be clear from the docs as it's there with example. But let me know if it isn't and we'll clarify this.